substrate/bin/node/runtime/src/lib.rs

```diff
@@ -745,6 +746,10 @@ impl pallet_staking::Config for Runtime {
 	type WeightInfo = pallet_staking::weights::SubstrateWeight<Runtime>;
 	type BenchmarkingConfig = StakingBenchmarkingConfig;
 	type DisablingStrategy = pallet_staking::UpToLimitWithReEnablingDisablingStrategy;
+	type MaxInvulnerables = ConstU32<20>;
+	type MaxRewardPagesPerValidator = ConstU32<20>;
+	type MaxValidatorsCount = ConstU32<300>;
```

Comment:
Other pallets like session, babe, grandpa and such also have a notion of `MaxValidators` that should be equal to the max validators of this pallet. 

within this file, you can use one `pub const MaxValidators: u32 = 300` everywhere to unify it. 

Taking it a step further, you can expose this as a part of `trait SessionManager`. this trait can expose a `trat SessionManager { type MaxAuthorities = <set-by-staking-pallet>`. Then, within the `pallet-session`, who consumes `SessionManager`, you can do 

```
fn integroty_check() { 
    /// A way to express within pallet-session that whoever implements `SessionManager` should have a compatible `MaxAuthorities`. 
    assert!(T::SessionManager::MaxAuthoritiet::get() >= T::MaxAuthorities::get())
}

```

This might be too much for this PR, but good for you to be familiar with the pattern. 

Whenever they are multiple parameters within two pallets that have a logical dependency (they have to be equal, or one has to be larger than the other), you can remove the implicitness like this.

Reply:
changed in https://github.com/paritytech/polkadot-sdk/pull/6445/commits/a065a91a59b019ab497e6b9748ed8fc925539f65

Opened issue #6972 to unify `MaxValidators` limit (I also see a `MaxActiveValidators`)